app/backend/app.py

```diff
@@ -106,8 +118,20 @@ async def content_file(path: str):
     try:
         blob = await blob_container_client.get_blob_client(path).download_blob()
     except ResourceNotFoundError:
-        logging.exception("Path not found: %s", path)
-        abort(404)
+        logging.info("Path not found in general Blob container: %s", path)
+        if current_app.config[CONFIG_USER_UPLOAD_ENABLED]:
+            try:
+                user_oid = auth_claims["oid"]
+                user_blob_container_client = current_app.config[CONFIG_USER_BLOB_CONTAINER_CLIENT]
+                user_directory_client: FileSystemClient = user_blob_container_client.get_directory_client(user_oid)
+                # Should we do an additional check on access control here?
```

Comment:
Is this an ADLS Gen2 container? If so we can add the actual ACL to the directory? It doesn't get automatically added to child files though unfortunately.

https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-acl-python

Reply:
We set the acl upon uploading in the upload route, later:

await user_directory_client.set_access_control(owner=user_oid)

Are you saying it should also be done here, in the content route?

Reply:
Got it - I just wanted to make sure that the ACL was set on the uploaded file.

Reply:
OK now I understand - my comment is in the wrong place